pkg/fish-spec/framework/assertions.fish

```diff
@@ -0,0 +1,189 @@
+    'Assertion failed: string \"$second\" does not match pattern \"$first\".' \
+    'string match -qr $first $second'
+end
+
+
+function assert_match -a pattern string
+  __fish_spec_assert_generic \
+    $pattern $string \
+    'Assertion string \"$string\" does not match pattern \"$pattern\" passed!' \
+    'Assertion failed: string \"$second\" does not match pattern \"$first\".' \
+    'not string match -qr $first $second'
+end
+
+function assert_file_exists -a file
+  __fish_spec_assert_generic \
+    $file unused \
+    'Assertion file \"$first\" exists passed!' \
+    'Assertion failed: File \"$first\" does not exist.' \
+    'test -f $first'
+end
+
+function assert_file_does_not_exist -a file
+  __fish_spec_assert_generic \
+    $file unused \
+    'Assertion file \"$first\" does not exist passed!' \
+    'Assertion failed: File \"$first\" exists.' \
+    'not test -f $first'
+end
+
+function assert_directory_exists -a dir
+  __fish_spec_assert_generic \
+    $dir unused \
+    'Assertion directory \"$first\" exists passed!' \
+    'Assertion failed: Directory \"$first\" does not exist.' \
+    'test -d $dir'
+end
+
+function assert_directory_does_not_exist -a file
+  __fish_spec_assert_generic \
+    $file unused \
+    'Assertion directory \"$first\" does not exist passed!' \
+    'Assertion failed: Directory \"$first\" exists.' \
+    'not test -d $first'
+end
+
+function assert_file_empty -a file
+  __fish_spec_assert_generic \
+    $file unused \
+    'Assertion file $first is empty passed!' \
+    'Assertion failed: File \"$first\" is not empty.' \
+    'not test -s $file'
+end
+
+function assert_file_contains -a file content
+  __fish_spec_assert_generic \
+    $file $content \
+    'Assertion $first contains \"$second\" passed!' \
+    'Assertion failed: File \"$first\" does not contain \"$second\".' \
+    'grep -q "$second" $first'
+end
+
+function assert_file_contains_regex -a file pattern
+  __fish_spec_assert_generic \
+    $file $pattern \
+    'Assertion $first content matches regex \"$second\" passed!' \
+    'Assertion failed: File \"$first\" does not contain a string matching the pattern \"$second\".' \
+    'grep -qE "$second" $first'
+end
+
+function assert_not_file_contains -a file content
+  __fish_spec_assert_generic \
+    $file $content \
+    'Assertion $first does not contain \"$content\" passed!' \
+    'Assertion failed: File \"$first\" contains \"$second\".' \
+    'not grep -q "$second" $first'
+end
+
+function assert_not_file_contains_regex -a file pattern
+  __fish_spec_assert_generic \
+    $file $pattern \
+    'Assertion $first content does not match regex \"$second\" passed!' \
+    'Assertion failed: File \"$first\" contains a string matching the pattern \"$second\".' \
+    'not grep -qE "$second" $first'
+end
+
+function __fish_spec_in_array_helper -a value
+  for item in $__fish_spec_assertion_array
+    if test "$item" = "$value"
+      return 0
+    end
+  end
```

Comment:
is there a particular reason to not use `contains` instead here?

Reply:
I didn't know `contains` existed :)
I've reworked my code to use `contains` instead. It's shorter and probably faster for large arrays